docs/en/sql-reference/functions/string-functions.md

```diff
@@ -2539,3 +2539,57 @@ Result:
 │ foo                        │
 └────────────────────────────┘
 ```
+## stringCompare
+
+Compare two strings lexicographically.
+
+**Syntax**
+
+```sql
+stringComare(string1, string2[, str1_off, string2_offset, num_bytes]);
+```
+
+**Arguments**
+
+- `string1` — The first string to compare. [String](../data-types/string.md)
+- `string2` - The second string to compare.[String](../data-types/string.md)
+- `string1_offset` — The position (zero-based) in `string1` from which the comparison starts. Optional, positive number.
+- `string2_offset` — The position (zero-based index) in `string2` from which the comparison starts. Optional, positive number.
+- `num_bytes` — The maximum number of bytes to compare in both strings. If `string_offset` + `num_bytes` exceeds the end of an input string, `num_bytes` will be reduced accordingly.
+
+**Returned value**
+
+- -1 — If `string1`[`string1_offset`: `string1_offset` + `num_bytes`] < `string2`[`string2_offset`:`string2_offset` + `num_bytes`] and `string1_offset` < len(`string1`) and `string2_offset` < len(`string2`).
+If `string1_offset` >= len(`string1`) and `string2_offset` < len(`string2`).
+- 0 — If `string1`[`string1_offset`: `string1_offset` + `num_bytes`] = `string2`[`string2_offset`:`string2_offset` + `num_bytes`] and `string1_offset` < len(`string1`) and `string2_offset` < len(`string2`).
+If `string1_offset` >= len(`string1`) and `string2_offset` >= len(`string2`).
+- 1 — If `string1`[`string1_offset`: `string1_offset` + `num_bytes`] > `string2`[`string2_offset`:`string2_offset` + `num_bytes`] and `string1_offset` < len(`string1`) and `string2_offset` < len(`string2`).
+If `string1_offset` < len(`string1`) and `string2_offset` >= len(`string2`).
+
+**Example**
+
+```sql
+SELECT
+    stringCompare('alice', 'bob', 0, 0, 3) as result1,
+    stringCompare('alice', 'alicia', 0, 0, 3) as result2,
+    stringCompare('bob', 'alice', 0, 0, 3) as result3
+```
+Result:
+```result
+   ┌─result1─┬─result2─┬─result3─┐
+1. │      -1 │       0 │       1 │
+   └─────────┴─────────┴─────────┘
+```
+
+```sql
```

Comment:
Let's order the simpler of the examples (this one) before the more complicated one.